module-bsp/board/rt1051/puretx/bsp/keyboard/keyboard.cpp

```diff
@@ -67,101 +77,140 @@ namespace bsp::keyboard
                 }
             }
 
-            gpio->EnableInterrupt(1U << static_cast<uint32_t>(BoardDefinitions::KEYBOARD_RF_BUTTON));
+            gpio->EnableInterrupt(1U << static_cast<std::uint32_t>(BoardDefinitions::KEYBOARD_RF_BUTTON));
         }
-
-        void clearAllIRQs()
-        {
-            std::uint8_t clearIRQs = 0xFF;
-            i2cAddr.subAddress     = REG_INT_STAT;
-            i2c->Write(i2cAddr, reinterpret_cast<std::uint8_t *>(&clearIRQs), 1);
-        }
-
     } // anonymous namespace
 
-    std::int32_t init(xQueueHandle qHandle)
+    void power(KeyboardPower state)
     {
+        gpio->WritePin(static_cast<std::uint32_t>(BoardDefinitions::KEYBOARD_RESET_PIN),
+                       static_cast<std::uint8_t>(state));
+    }
 
-        i2c = DriverI2C::Create(
-            static_cast<I2CInstances>(BoardDefinitions::KEYBOARD_I2C),
-            DriverI2CParams{.baudrate = static_cast<uint32_t>(BoardDefinitions::KEYBOARD_I2C_BAUDRATE)});
-
-        gpio = DriverGPIO::Create(static_cast<GPIOInstances>(BoardDefinitions::KEYBOARD_GPIO), DriverGPIOParams{});
-
-        qHandleIrq = qHandle;
+    void reset()
+    {
+        // According to datasheet, section 6.7
+        power(KeyboardPower::off);
+        // Reset pulse should be at least 120us
+        vTaskDelay(pdMS_TO_TICKS(chipResetMs));
+        power(KeyboardPower::on);
+        // Recovery time is 120us
+        vTaskDelay(pdMS_TO_TICKS(chipRecoveryMs));
+    }
 
-        gpio->ConfPin(DriverGPIOPinParams{.dir      = DriverGPIOPinParams::Direction::Input,
-                                          .irqMode  = DriverGPIOPinParams::InterruptMode::IntRisingOrFallingEdge,
-                                          .defLogic = 0,
-                                          .pin      = static_cast<uint32_t>(BoardDefinitions::KEYBOARD_RF_BUTTON)});
+    bool write(const drivers::I2CAddress &addr, const std::uint8_t *txBuff, const std::size_t size)
```

Comment:
just wondering - wouldn't it look nicer with returning `std::pair` with ret code and buffer pointer instead of C-style parameter passing?
The same goes to `read`

It's only a suggestion - if you don't find it useful - feel free to resolve it anytime :D

Reply:
Thanks for your suggestion. It is a good idea but most drivers use this same pattern of parameter passing. So I leave it now as it is.